### PR TITLE
Redacts sensitive/password information when listing streams

### LINKED ISSUE #1542: Password should be hidden in UI/shell
_From @kniemann on May 11, 2017 21:1_

Passwords in the Dataflow UI should be hidden. This is probably true in the REST API/shell as well. Previously in SpringXD they were automatically ****. Especially since a VIEW permission was added in 1.2.0, it wouldn't make sense to grant it if passwords are visible in plain text.

**SpringXD:**

image

**vs Spring Cloud:**

image

_Copied from original issue: spring-cloud/spring-cloud-dataflow-ui#215_

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #1542

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java [hunk 1]
```diff
@@ -18,13 +18,17 @@
 
 import java.util.regex.Pattern;
 
+import org.springframework.cloud.dataflow.core.dsl.ArgumentNode;
+
 /**
  * Sanitizes potentially sensitive keys for a specific command line arg.
  * @author Glenn Renfro
  */
 public class ArgumentSanitizer {
```
Comment (markpollack) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java]: Do you happen to know why the boot guys didn't make this class public

https://pivotal.slack.com/archives/G0M0F86AJ/p1497984170701058

Comment (markpollack) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java]: I can't find it ATM, but I think there is a way in boot to modify which keys are considered the ones to sanitize.... @trisberg recently was asking how get boot to give up it's secrets.....   Did you find anything along that line of thought?

Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java]: I saw their implementation.  But it is not accessible.  So this is based on their implementation.

Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/ArgumentSanitizer.java]: Did you mean this ?
https://stackoverflow.com/questions/28300232/spring-boot-actuator-hides-property-values-in-env-endpoint

**Comment by cppwfs** [comment]
Closing this PR.  Please use: https://github.com/spring-cloud/spring-cloud-dataflow/pull/1561/files

**Pull request closed without merging**